base_wamas_ubl/lib/wamas/utils.py

```diff
@@ -211,9 +211,16 @@ def get_date_from_field(*args):
     return res
 
 
-def convert_unit_code(key, val):
+def convert_unit_code(key, val, unit_mapping=None):
     if key in LST_FIELD_UNIT_CODE:
-        return MAPPING_UNITCODE_UBL_TO_WAMAS["unitCode"].get(val, val)
+        MAPPING = (
+            unit_mapping
+            and unit_mapping.get(
+                "MAPPING_UNITCODE_UBL_TO_WAMAS", MAPPING_UNITCODE_UBL_TO_WAMAS
+            )
+            or MAPPING_UNITCODE_UBL_TO_WAMAS
+        )
+        return MAPPING["unitCode"].get(val, val)
```

Comment:
Here `MAPPING` doesn't refer to a "constant" object at the module level, better to write it in lowercase:
```suggestion
        _mapping = (
            unit_mapping
            and unit_mapping.get(
                "MAPPING_UNITCODE_UBL_TO_WAMAS", MAPPING_UNITCODE_UBL_TO_WAMAS
            )
            or MAPPING_UNITCODE_UBL_TO_WAMAS
        )
        return _mapping["unitCode"].get(val, val)
```